### PR TITLE
Support for disabling static product dependencies linting

### PR DESCRIPTION
**Pull request by danyf90**
Properly fix automatic resource warnings: the previous workaround didn't work with local packages
Support disabling static product lint warnings, as in some case it's fine to have such dependencies, and having too many warnings might hide other important ones

### CODE REVIEW

File: Sources/TuistGenerator/Linter/GraphLinter.swift [hunk 1]
```diff
@@ -5,7 +5,7 @@ import TuistGraph
 import TuistSupport
 
 public protocol GraphLinting: AnyObject {
-    func lint(graphTraverser: GraphTraversing) -> [LintingIssue]
+    func lint(graphTraverser: GraphTraversing, disableStaticProductsLint: Bool) -> [LintingIssue]
 }
```
Comment (kwridan) [inline Sources/TuistGenerator/Linter/GraphLinter.swift:9]: Seeing this is a workspace generation option, perhaps can be extracted from `graphTraverser.workspace`?

This should reduce the necessary code changes 😅

**Review by kwridan — COMMENTED** [review]
Thanks @danyf90 !

Would be curious to understand more about the cases where duplicate static products occur.

**Comment by danyf90** [comment]
> Would be curious to understand more about the cases where duplicate static products occur.

AFAIK having a lib linked twice is not necessarily a problem unless the library does some work at load/link time. Linking them as static in this case might reduce the app size and load time.

Also the problem might arise for the same library linked in two separate part of the project tree (e.g. the main app and a feature demo app)

**Review by kwridan — APPROVED** [review]
Thanks @danyf90 🙏

> AFAIK having a lib linked twice is not necessarily a problem unless the library does some work at load/link time. Linking them as static in this case might reduce the app size and load time.

There are other very subtle issues that arise where clients of the static library expect types within it to be the same from different contexts / targets - e.g. `ObjectIdentifier`s of objects will differ, and if there are any singletons ... they won't be anymore 😅

I was mostly curious as what motivated this change - as I also see some unrelated changes regarding resources in dependencies. It would be a shame to opt users out of this warning if there is a different underlying issue.
 
> Also the problem might arise for the same library linked in two separate part of the project tree (e.g. the main app and a feature demo app)

Worth flagging this, as I recall we attempted to fix this issue in a past iteration of this linter

 e.g. 

- https://github.com/tuist/tuist/pull/981
- https://github.com/tuist/tuist/pull/2201

Testing against our project where we following this pattern - I don't see any false positives 🤔

I wonder if later on we would assign a unique ID for each of the lint rules to allow users to opt in / out of specific rules or even elevate their severity (e.g. https://github.com/tuist/tuist/issues/4376)

**Pull request closed without merging**